pandas/core/internals/managers.py

```diff
@@ -1339,16 +1339,15 @@ def _slice_take_blocks_ax0(self, slice_or_indexer, fill_tuple=None):
             if sl_type in ("slice", "mask"):
                 return [blk.getitem_block(slobj, new_mgr_locs=slice(0, sllen))]
             elif not allow_fill or self.ndim == 1:
-                if allow_fill and fill_tuple[0] is None:
+                if allow_fill and fill_value is None:
```

Comment:
is allow_fill now redundant here?

Reply:
allow_fill is tricky to remove (but this is a start to it)

Reply:
The `allow_fill` is from checking `fill_value is not lib.default`. So `fill_value is None` is still a different check I think.

Reply:
what @jorisvandenbossche said

Reply:
if `fill_value is None`, isn't `allow_fill`, by definition, `True`. so `allow_fill` is redundant here and `if fill_value is None:` would be the same? nbd though, i must be missing something. will look again. We no longer need the guard, that was previously required with the `fill_tuple` \_\_getitem__ call.